packages/router/src/__tests__/set.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { render } from '@testing-library/react'
```

Comment:
Nice one @Tobbe. Do you think we could add some tests here for providers/contexts? I'm not exactly sure how we'd test that, but I expect it'll be very useful to prevent regression

Reply:
That's a great idea! I went ahead and added it to the router tests. Please take a look